assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * DashboardSummaryWidget component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { reduceAdSenseData } from '../../util';
+import { readableLargeNumber, extractForSparkline, getSiteKitAdminURL } from '../../../../util';
+import whenActive from '../../../../util/when-active';
+import PreviewBlock from '../../../../components/preview-block';
+import Layout from '../../../../components/layout/layout';
+import DataBlock from '../../../../components/data-block';
+import Sparkline from '../../../../components/sparkline';
+import getDataErrorComponent from '../../../../components/notifications/data-error';
+import getNoDataComponent from '../../../../components/notifications/nodata';
+
+const { useSelect } = Data;
+
+function DashboardSummaryWidget() {
+	const {
+		error,
+		loading,
+		today,
+		period,
+		daily,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+		const metrics = [ 'EARNINGS', 'PAGE_VIEWS_RPM', 'IMPRESSIONS' ];
+
+		const format = 'YYYY-MM-DD';
+		const now = moment().format( format );
+		const lastMonth = moment().subtract( 1, 'month' ).format( format );
+
+		const todayArgs = {
+			startDate: now,
+			endDate: now,
+			metrics,
+		};
+
+		const periodArgs = {
+			dateRange: select( CORE_USER ).getDateRange(),
+			metrics,
+		};
+
+		const dailyArgs = {
+			startDate: lastMonth,
+			endDate: now,
+			metrics,
+			dimensions: [ 'DATE' ],
+		};
```

Comment:
Introducing a new dependency here and migrating to pass dates directly goes a bit beyond the scope of this one. Let's instead rely on the `dateRange` slugs that are present in the legacy version of the widget (e.g. `today` and `this-month`). The AdSense module class in PHP has extra logic to support those.